sqlite_query.c

```diff
@@ -44,52 +52,242 @@ sqlite_convert_to_pg(Oid pgtyp, int pgtypmod, sqlite3_stmt * stmt, int stmt_coli
+						{
+							int			value = sqlite3_column_int(stmt, stmt_colid);
+							return (struct NullableDatum){BoolGetDatum(value), false};
+						}
+					case SQLITE_FLOAT:
+					case SQLITE_BLOB:
+						{
+							sqlite_value_to_pg_error (pgtyp, pgtypmod, stmt, attnum, sqlite_value_affinity, affinity_for_pg_column, value_byte_size_blob_or_utf8);
+							break;
+						}
+					case SQLITE3_TEXT:
+					{
+						if (value_byte_size_blob_or_utf8)
+							sqlite_value_to_pg_error (pgtyp, pgtypmod, stmt, attnum, sqlite_value_affinity, affinity_for_pg_column, value_byte_size_blob_or_utf8);
+						else
+							elog(ERROR, "Void text disallowed for PostgreSQL bool column");
+						break;
+					}
+					default:
+					{
+						sqlite_value_to_pg_error (pgtyp, pgtypmod, stmt, attnum, sqlite_value_affinity, affinity_for_pg_column, value_byte_size_blob_or_utf8);
+						break;
+					}
+				}
+				break;
+			}
 		case BYTEAOID:
 			{
-				// int			value_byte_size_blob_or_utf = sqlite3_column_bytes(stmt, stmt_colid); // Calculated always for detectind void values
-				value_datum = (Datum) palloc0(value_byte_size_blob_or_utf8 + VARHDRSZ);
-				memcpy(VARDATA(value_datum), sqlite3_column_blob(stmt, stmt_colid), value_byte_size_blob_or_utf8);
-				SET_VARSIZE(value_datum, value_byte_size_blob_or_utf8 + VARHDRSZ);
-				return (struct NullableDatum) { PointerGetDatum(value_datum), false};
+				switch (sqlite_value_affinity)
+				{
+					case SQLITE_INTEGER:
+					case SQLITE_FLOAT:
+						{
+							sqlite_value_to_pg_error (pgtyp, pgtypmod, stmt, attnum, sqlite_value_affinity, affinity_for_pg_column, value_byte_size_blob_or_utf8);
+							break;
+						}
+					case SQLITE_BLOB: /* <-- proper and recommended SQLite affinity of value for pgtyp */
+					case SQLITE3_TEXT: /* threated as UTF-8 text BLOB */
+					{
+						value_datum = (Datum) palloc0(value_byte_size_blob_or_utf8 + VARHDRSZ);
+						memcpy(VARDATA(value_datum), sqlite3_column_blob(stmt, stmt_colid), value_byte_size_blob_or_utf8);
+						SET_VARSIZE(value_datum, value_byte_size_blob_or_utf8 + VARHDRSZ);
+						return (struct NullableDatum) {PointerGetDatum(value_datum), false};
+					}
+					default:
+					{
+						sqlite_value_to_pg_error (pgtyp, pgtypmod, stmt, attnum, sqlite_value_affinity, affinity_for_pg_column, value_byte_size_blob_or_utf8);
+						break;
+					}
+				}
+				break;
 			}
 		case INT2OID:
 			{
-				int			value = sqlite3_column_int(stmt, stmt_colid);
-
-				return (struct NullableDatum) { Int16GetDatum(value), false};
+				switch (sqlite_value_affinity)
+				{
+					case SQLITE_INTEGER: /* <-- proper and recommended SQLite affinity of value for pgtyp */
+						{
+							int			value = sqlite3_column_int(stmt, stmt_colid);
+							return (struct NullableDatum) {Int16GetDatum(value), false};
+						}
+					case SQLITE_FLOAT:
+						{
+							sqlite_value_to_pg_error (pgtyp, pgtypmod, stmt, attnum, sqlite_value_affinity, affinity_for_pg_column, value_byte_size_blob_or_utf8);
```

Comment:
Similar with default case. Please check for other places also.

Reply:
Fixed, thanks.

Reply:
Thank for updating. Confirmed.